sdcm/cluster.py

```diff
@@ -605,6 +609,37 @@ def cpu_cores(self):
         cores = int(cores) if cores else None
         return cores
 
+    @property
+    def scylla_shards(self):
+        if not self._scylla_shards:
+            # Possible output of grep:
+            #   'CPUSET="--cpuset 1-7 "'
+            #   'CPUSET="--cpuset 1-7,9-15 "'
+            grep_result = self.remoter.run('grep "^CPUSET" /etc/scylla.d/cpuset.conf')
+            scylla_cpu_set = re.findall(r'(\d+)', grep_result.stdout)
+            self.log.debug(f"CPUSET on node {self.name}: {grep_result}")
+
+            if not scylla_cpu_set:
+                self.log.error(f'Failed to parse CPUSET from "/etc/scylla.d/cpuset.conf". '
+                               f'Unexpected CPUSET is "{grep_result}"')
+                self._scylla_shards = self.cpu_cores
+
+            self._scylla_shards = int(scylla_cpu_set[1]) - int(scylla_cpu_set[0]) + 1
+
+            if len(scylla_cpu_set) == 4:
+                self._scylla_shards += int(scylla_cpu_set[3]) - int(scylla_cpu_set[2]) + 1
+
+        return self._scylla_shards
+
+    @cached_property
+    def scylla_random_shards(self):
```

Comment:
convert to method, we need to have a single point of Truth

Reply:
done